ownCloud/Client/Actions/Actions+Extensions/DuplicateAction.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  DuplicateAction.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 15/11/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import Foundation
+
+import ownCloudSDK
+
+class DuplicateAction : Action {
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.duplicate") }
+	override class var category : ActionCategory? { return .normal }
+	override class var name : String? { return "Duplicate".localized }
+
+	// MARK: - Extension matching
+	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
+		// Examine items in context
+		return .middle
+	}
+
+	// MARK: - Action implementation
+	override func run() {
+		guard context.items.count > 0 else {
+			completionHandler?(NSError(ocError: .errorInsufficientParameters))
+			return
+		}
+
+		beforeRunHandler?()
+
+		let item = context.items[0]
+		let rootItem = context.items[1]
+
+		var name: String = "\(item.name!) copy"
+
+		if item.type != .collection {
+			let itemName = item.nameWithoutExtension()
+			var fileExtension = item.fileExtension()
+
+			if fileExtension != "" {
+				fileExtension = ".\(fileExtension)"
+			}
+
+			name = "\(itemName) copy\(fileExtension)"
```

Comment:
This will fail when trying to make a second copy without renaming the first one before.

Makes me wonder… should the `OCCore` provide a method to suggest a likely unused, numbered variant of an item's name? Something like `core.suggestUnusedName(for: item, withAppendedLabel: "copy")`?

Reply:
It could be one approach, another option is that the SDK takes care itself about the naming, so call `duplicate` function and the SDK provides the name, create the duplicate file and call the query delegate with the new item.